tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/RecordMatcher.cs

```diff
@@ -38,6 +38,7 @@ public RecordMatcher(bool compareBodies = true)
             "x-ms-date",
             "x-ms-client-request-id",
             "User-Agent",
+            "x-ms-useragent",
```

Comment:
marking more exclusions to consider
![image](https://user-images.githubusercontent.com/10452642/142297780-7e197ad4-3041-4499-97d3-0210a1a1d12b.png)